src/components/CartCounter/CartCounter.tsx

```diff
@@ -4,31 +4,67 @@ import Grid from "@mui/material/Unstable_Grid2";
 import { IconButton } from "@mui/material";
 import { Add, Remove } from "@mui/icons-material";
 
+import { useCart } from "../../pages/CartPage/CartContext";
+
 import styles from "./CartCounter.module.scss";
 
-const CartCounter = () => {
-  const [amount, setAmount] = useState(1);
-  const plusAmount = () => {
-    setAmount(amount + 1);
+interface CartCounterInterface {
+  id: number;
+  amount: number;
+  setAmountBeforeAddToCart?: React.Dispatch<React.SetStateAction<number>>;
+}
+
+const CartCounter = ({
+  id,
+  amount,
+  setAmountBeforeAddToCart,
+}: CartCounterInterface) => {
+  const { cart, increaseAmount, decreaseAmount } = useCart();
+
+  const [innerAmount, _] = useState(amount);
+  const isProductInCart = !!cart.find((product) => product.id === id);
```

Comment:
You can use `Array.some` for this
```suggestion
  const isProductInCart = !!cart.find((product) => product.id === id);
```